src/utils/safeHref.ts

```diff
@@ -0,0 +1,90 @@
+export interface URLInfo {
+    url: string | undefined
+    sameOrigin: boolean
+}
+
+let useSafeHref = false
+
+export function setUseSafeHref(value: boolean) {
+    useSafeHref = value
+}
+
+export function getUseSafeHref(): boolean {
+    return useSafeHref
+}
+
+export function getURLInfo(url: string): URLInfo {
+    /**
+     * An array of the safely allowed url protocols
+     */
+    const safeProtocols = ['http:', 'https:', 'mailto:', 'tel:', 'data:']
+
+    /**
+     * Create an anchor element to parse the protocol and origin of the url provided
+     */
+    const anchorElement = document.createElement('a')
+    anchorElement.href = url
+
+    /**
+     * Compare window location if not ssr
+     */
+    const sameOrigin = typeof window !== undefined ? anchorElement.origin === window.location.origin : true
```

Comment:
Maybe we should provide a way to manually configure the origin? Because then the server and client side can have consistent render results.

Reply:
Yeah, seems reasonable. Perhaps part of `setUseSafeHref` is being able to specify the origin.

Reply:
I think it should be a separate method or it should be renamed to `configureSafeHref` or create a generic config method for all of ui-box.

Reply:
Makes sense! wdyt @colinlohner ?